c10/cuda/CUDACachingAllocator.cpp

```diff
@@ -3263,9 +3263,9 @@ static bool forceUncachedAllocator() {
   // Allow either CUDA or HIP name for env var for maximum user comfort
   // the CUDA env var avoids being hipified in cuda_to_hip_mappings.py
   static bool has_cuda_env =
-      c10::utils::has_env("PYTORCH_NO_CUDA_MEMORY_CACHING");
+      c10::utils::get_env("PYTORCH_NO_CUDA_MEMORY_CACHING") == "1";
```

Comment:
only nit is to reuse check_env logic similar to:
```suggestion
      c10::utils::check_env("PYTORCH_NO_CUDA_MEMORY_CACHING");
```